includes/class-donations.php

```diff
@@ -848,5 +849,26 @@ public static function can_use_streamlined_donate_block() {
 		}
 		return false;
 	}
+
+	/**
+	 * Manipulate WC's cart, if needed.
+	 * If WC is not the donations platform, the donation products should not be buyable.
+	 */
+	public static function handle_cart() {
+		if ( ! self::is_woocommerce_suite_active() ) {
+			return;
+		}
+		if ( self::is_platform_wc() ) {
+			return;
+		}
+		if ( \is_cart() || \is_checkout() ) {
```

Comment:
Since this is running in every request, I was wondering what would be the way to impact the less on performance.

I guess if we first checked for `is_cart() || is_checkout` and do the other 2 checks after that we will save some processing cycles, wouldn't we? we would only check those things if we are on the right page... WDYT?

Reply:
I'm not sure, actually. `is_cart` and `is_checkout` are the "heaviest" checks – both read an option and check for shortcodes in page, among other things. On the other hand, if we assume that most sites have WC active, the first check would be passed in most cases. Overall, I don't have a strong opinion on this kind of micro-optimisation.

Reply:
yeah it's hard to know. Best case scenario would be to use a hook that is fired only on checkout and cart pages. Have you considered using `woocommerce_before_checkout_form` and `woocommerce_before_cart`?

I just like to avoid having thins running in each and every request, but feel free to go ahead and merge it if you don't think this is a problem

Reply:
Great suggestion! Done in 27e8095fc87aadb59fa2e46ba655fa27d1fc0262